bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -238,6 +239,14 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
         return codec;
     }
 
+    private <S> Codec<S> getCodecFromPropertyRegistry(final PropertyModel<S> propertyModel) {
+        try {
+            return propertyCodecRegistry.get(propertyModel.getTypeData());
```

Comment:
If there is no `Codec` available instead of throwing an exception immediately, an exception will be thrown lazily only if that propertyModel is encoded / decoded.